### PR TITLE
STM32MP1 Support

### PR DESCRIPTION
**Pull request by EmbeddedAndroid**
Initial changes for meta-lmp to support the family of STM32MP1 SoCs.

### CODE REVIEW

File: meta-lmp-bsp/recipes-bsp/u-boot/u-boot-stm32mp/0001-include-configs-stm32mp1-set-lmp-boot-cmd.patch [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+-		"run env_check;" \
+-		"if test ${boot_device} = mmc;" \
+-		"then env set boot_targets \"mmc${boot_instance}\"; fi;" \
+-		"if test ${boot_device} = nand ||" \
+-		  " test ${boot_device} = spi-nand ;" \
+-		"then env set boot_targets ubifs0; fi;" \
+-		"if test ${boot_device} = nor;" \
+-		"then env set boot_targets mmc0; fi;" \
+-		"run distro_bootcmd;" \
++		"setenv devtype mmc ; setenv devnum 1 ; setenv fdtfile stm32mp157c-ev1.dtb ; setenv addr_fit 0xC0008000 ;" \
```
Comment (EmbeddedAndroid) [inline meta-lmp-bsp/recipes-bsp/u-boot/u-boot-stm32mp/0001-include-configs-stm32mp1-set-lmp-boot-cmd.patch:32]: Need to find a way to get the lmp.cfg to work with u-boot-stm32mp then we can define the fdt variables per machine.

**Comment by EmbeddedAndroid** [comment]
One missing thing, is I need to figure out a way to ship this FlashLayout text file

`#Opt	Id	Name	Type	IP	Offset	Binary
-	0x01	fsbl1-boot	Binary	none	0x0	tf-a-stm32mp157c-ev1-serialboot.stm32
-	0x03	ssbl-boot	Binary	none	0x0	u-boot-stm32mp157c-ev1-trusted.stm32
P	0x04	fsbl1	Binary	mmc1	boot1	tf-a-stm32mp157c-ev1-trusted.stm32
P	0x05	fsbl2	Binary	mmc1	boot2	tf-a-stm32mp157c-ev1-trusted.stm32
PD	0x06	ssbl	Binary	mmc1	0x00080000	u-boot-stm32mp157c-ev1-trusted.stm32
P	0x21	otaroot	FileSystem	mmc1	0x00280000	lmp-factory-image-stm32mp1-eval.ext4`

**Comment by EmbeddedAndroid** [comment]
Ok, this is ready for review I think. Here are the relevant PRs for this platform support:

- https://github.com/foundriesio/lmp-kernel-cache/pull/6

- https://github.com/foundriesio/lmp-manifest/pull/111

Once we merge the above, I'll update the kmeta references.

**Comment by ricardosalveti** [comment]
Merged changes by hand as I moved them to bblayers-dynamic, so they can only be available in case meta-stm is also available and enabled by bblayers.

**Pull request closed without merging**